core/src/swarm/registry.rs

```diff
@@ -62,17 +66,41 @@ impl Addresses {
     /// Adding an existing address is interpreted as additional
     /// confirmation and thus increases its score.
     pub fn add(&mut self, a: Multiaddr) {
+        debug_assert_ne!(self.reports.capacity(), 0);
+
+        let oldest = if self.reports.len() == self.reports.capacity() {
```

Comment:
I suggest to add a separate field instead of using the capacity because the value passed to `VecDeque::with_capacity` may not be the actual capacity in use which makes it harder to understand the program. A field `max_history: usize` would be more straightforward.